### PR TITLE
docs(deps): up sphinx to 3.4.3, add napoleon_attr_annotations=True

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1665 

Description: 
- upgrade sphinx to 3.4.3
- add `napoleon_attr_annotations=True` in conf.py

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -206,8 +206,10 @@
 
 # -- Type hints configs ------------------------------------------------------
 
+autodoc_inherit_docstrings = True
```
Comment (vfdev-5) [inline docs/source/conf.py:209]: Default is also True, but let's keep it like that

https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html

**Review by vfdev-5 — APPROVED** [review]
Thanks @ydcjeff !

File: docs/source/conf.py (hunk 1 above)
Comment (ydcjeff) [reply docs/source/conf.py:209]: Yes, we want to have `True` value, but keeping it just in case sphinx change to `False`

**Comment by vfdev-5** [comment]
It's funny that the preview does not show
```
Notes[ + ]

Package Reference[ + ]

Contrib Package Reference[ + ]

Team[ + ]
```
the way how the docs lookk like now. @ydcjeff any ideas on how to keep the previous rendering with Package Reference, Contrib Package Reference and Team always expanded ?

**Comment by ydcjeff** [comment]
I think it is related to the recent changes of `pytorch_sphinx_theme`

**Comment by vfdev-5** [comment]
@ydcjeff so should we do something with it ?

**Pull request merged**